test/factories/providers.rb

```diff
@@ -33,24 +33,32 @@
       discarded_at { Time.zone.now }
     end
 
-    trait :unaccredited do
-      accreditation_status { :unaccredited }
-      provider_type { ProviderTypeEnum::UNACCREDITED_PROVIDER_TYPES.keys.sample }
+    trait :school do
+      provider_type { :school }
     end
 
-    trait :accredited do
-      accreditation_status { :accredited }
-      provider_type { ProviderTypeEnum::ACCREDITED_PROVIDER_TYPES.keys.sample }
+    trait :scitt do
+      provider_type { :scitt }
     end
 
-    trait :school do
-      provider_type { :school }
+    trait :hei do
+      provider_type { :hei }
+    end
+
+    trait :other do
+      provider_type { :other }
+    end
+
+    trait :unaccredited do
       accreditation_status { :unaccredited }
     end
```

Comment:
There needs to be a constraint here as a 
- `school` when accredited is a `scitt` 
- `sciitt` when unaccredited is a `school`